### ISSUE TITLE
New modules for RHEL8 Summit upgrade

### ISSUE THREAD

**Issue by amametjanov** [E3SM-Project/E3SM#4458]
>RHEL8 Upgrade on Summit (Aug 17-19)
Summit will be unavailable from 8:00 AM on Tuesday, August 17 until 8:00 AM on Thursday, August 19. During this time the OS will be upgraded to RHEL 8. Software changes related to this upgrade (both the available modules and the default versions) are noted on the [Software News](https://docs.olcf.ornl.gov/software/software-news.html) page. We strongly recommend that you recompile any codes you will run on the upgraded system following the outage. If you have any questions or concerns about this upgrade, contact the User Assistance Center at help@olcf.ornl.gov.

Modules used by E3SM before the upgrade:
ibmgpu
```
Currently Loaded Modules:
  1) hsi/5.0.2.p5            6) python/3.7.0      11) xl/16.1.1-9                     16) netcdf-fortran/4.4.4
  2) xalt/1.2.1              7) subversion/1.9.3  12) spectrum-mpi/10.3.1.2-20200121  17) parallel-netcdf/1.8.1
  3) lsf-tools/2.0           8) git/2.13.0        13) netlib-lapack/3.8.0             18) hdf5/1.10.4
  4) darshan-runtime/3.2.1   9) cmake/3.20.2      14) cuda/10.1.243
  5) DefApps                10) essl/6.1.0-2      15) netcdf/4.6.1
```
pgigpu
```
Currently Loaded Modules:
  1) hsi/5.0.2.p5            5) DefApps            9) cmake/3.20.2       13) cuda/10.1.243                   17) netcdf-fortran/4.4.4
  2) xalt/1.2.1              6) python/3.7.0      10) essl/6.1.0-2       14) netlib-lapack/3.8.0             18) parallel-netcdf/1.8.1
  3) lsf-tools/2.0           7) subversion/1.9.3  11) pgi/19.9           15) spectrum-mpi/10.3.1.2-20200121  19) hdf5/1.10.4
  4) darshan-runtime/3.2.1   8) git/2.13.0        12) pgi-cxx14/default  16) netcdf/4.6.1
```
gnugpu
```
Currently Loaded Modules:
  1) hsi/5.0.2.p5            5) DefApps            9) essl/6.1.0-2   13) netlib-lapack/3.8.0             17) parallel-netcdf/1.8.1
  2) xalt/1.2.1              6) python/3.7.0      10) cuda/10.1.105  14) spectrum-mpi/10.3.1.2-20200121  18) hdf5/1.10.4
  3) lsf-tools/2.0           7) subversion/1.9.3  11) gcc/8.1.1      15) netcdf/4.6.1
  4) darshan-runtime/3.2.1   8) cmake/3.20.2      12) git/2.13.0     16) netcdf-fortran/4.4.4
```
All three compiler + cuda modules are going away.

**Comment by rljacob** [comment]
Attention @philipwjones

**Comment by amametjanov** [comment]
I think we'll be fine with new default `xl/16.1.1-10`.
For gcc, @mrnorman checked that `gcc/9.3.0`+`cuda/11.4.0` works.
For pgi, 19.9 had run-time errors (Compy went up to `pgi/19.10` because of it). If 20.1 or 20.4 don't work, we have to ask for 19.10.

**Comment by minxu74** [comment]
Okay. I will test PGI/20.1 or 20.4 after Summit is up, Thanks

**Comment by minxu74** [comment]
I checked the Summit system this morning. It seemed that for PGI and GNU, the old compiler and cuda versions needed by E3SM are still available. For the XL compiler, only the new version is available now, but the old cuda version is still available.  For all compilers, the spectrum-mpi is gone.

**Comment by amametjanov** [comment]
We need to get to the same pass/fail rate as before the upgrade. Here are some baselines before the OS update: from https://my.cdash.org/index.php?project=E3SM&date=2021-08-14 .

gnu (4/91 fail/pass): https://my.cdash.org/viewTest.php?buildid=2052207
gnugpu (6/89): https://my.cdash.org/viewTest.php?buildid=2052257
ibm (6/90): https://my.cdash.org/viewTest.php?buildid=2052100
ibmgpu (18/77): https://my.cdash.org/viewTest.php?buildid=2052213

**Comment by minxu74** [comment]
@amametjanov thanks a lot for pointing out the baseline. I will do the integration tests and compared them with the baseline. By the way, where are the baselines for pgi and pgignu?

**Comment by amametjanov** [comment]
To clarify, I don't think we need BFB baseline comparison, just the same pass rate. :)
Based on spot-check tests, it didn't look like `pgi/19.9` was working: I had to go up to 19.10 to get `SMS_D_Ln5.ne4_ne4.FC5AV1C-L.summit_pgi` running without errors.

**Comment by minxu74** [comment]
@amametjanov did you also remove the pgi-cxx too? the pgi/19.10 is gone on Summit too. I plan to update it to the latest 20.4 if there is no problem with this version

**Comment by amametjanov** [comment]
To get newer gcc CXX libs (what `pgi-cxx` was doing), here is what I did on Ascent:
```
module use /gpfs/wolf/cli115/world-shared/e3sm/tools
module load pgirc/cxx14
export NVCC_WRAPPER_DEFAULT_COMPILER=pgc++
```
where that module is just
```
[azamat@login1.ascent ~]$ cat /gpfs/wolf/cli115/world-shared/e3sm/tools/pgirc/cxx14.lua 
whatis([[Name : pgi-cxx14 RC]])
whatis([[Version : 20.1]])
whatis([[Short description : local RC file for gcc with c++14.]])

help([[Local RC file for gcc with c++14.]])

-- Loading this module unlocks the path below unconditionally
setenv("PGI_LOCALRC", "/gpfs/wolf/cli115/world-shared/e3sm/tools/pgirc/pgi-19.10-rc")
```
and a new RC file can be created with PGI's `makelocalrc`.

**Comment by minxu74** [comment]
Some updates on nvhpc in the issue #4479

**Comment by amametjanov** [comment]
Fixed by #4502 and #4863 .
